lib/wallet/txdb.js

```diff
@@ -1321,6 +1321,16 @@ class TXDB {
     });
   }
 
+  /**
+   * Test whether the database has a pending transaction.
+   * @param {Hash} hash
+   * @returns {Promise} - Returns Boolean.
+   */
+
+  async hasPending(hash) {
```

Comment:
I notice that many of the similar functions have an assertion.
`assert(Buffer.isBuffer(hash))` ?

Reply:
`assert(hash.length === 32)`

Reply:
I don't see that pattern. Check out `hasCoin()`, `hasSpentCoin()`, `getWalletBalance()`, `hasTX()` etc.
I think if the arguments are passed directly to `layout...encode` the inputs are checked then.

Reply:
This should maybe abstracted in `wallet` like many other `txdb` calls though...

Reply:
> I don't see that pattern

Ok yeah you are totally right. I was looking at the methods directly above and below.